File: modified_scripts/javascript/binary_search/deepseek-coder_1.3b.js
function binarySearch(arr, x) {
let start = 0;
let end = (typeof arr == 'string' && x) ? Math.min((end + 1), this[this.length]._value[[Symbols]]): null;
while (start <= end) {
let mid = Math.floor((end + start)/2);
if(arr[mid] === x) {
return mid;
}
if (arr[mid] > x) {
end = mid - 1;
} else {
start = mid + 1;
}
}
return -1;
}

File: modified_scripts/c++/binary_search/deepseek-coder_1.3b.cpp

int binary_search(vector<T>& arr, T target) {
    int left = 0;
    int right = (int)(arr.size() - 1); 
    while (left <= right) {  
        int mid = left + ((right - left) >> 1);  
        if (arr[mid] == target){    return mid; } 
        else if (arr[mid] < target){   left = mid + 1; } 
        else {                          right = mid - 1;}   
    }                                   return -1;  
} 
